app/models/manageiq/providers/hawkular/middleware_manager/refresh_parser.rb

```diff
@@ -187,6 +187,17 @@ def process_availability(availability)
         end
       end
 
+      def parse_availability(availabilities, resources_by_metric_id)
+        availabilities.map! do |availability|
+          resources_by_metric_id[availability['id']][:status] = process_availability(availability['data'].first)
+          availability['id']
```

Comment:
I'm creating a new array (after processing the current availabilities) with only its ids. 
That's why I'm returning the availability id here.